test/kubernetes/e2e/test.go

```diff
@@ -314,6 +315,59 @@ func (i *TestInstallation) PreFailHandler(ctx context.Context) {
 	kubectlGetResourcesCmd := i.Actions.Kubectl().Command(ctx, "get", strings.Join(resourcesToGet, ","), "-A", "-owide")
 	_ = kubectlGetResourcesCmd.WithStdout(clusterStateFile).WithStderr(clusterStateFile).Run()
 	clusterStateFile.WriteString("\n")
+
+	podStdOut := bytes.NewBuffer(nil)
+	podStdErr := bytes.NewBuffer(nil)
+
+	// Fetch the name of the Gloo Gateway controller pod
+	getGlooPodNameCmd := i.Actions.Kubectl().Command(ctx, "get", "pod", "-n", i.Metadata.InstallNamespace,
+		"--selector", "gloo=gloo", "--output", "jsonpath='{.items[0].metadata.name}'")
+	_ = getGlooPodNameCmd.WithStdout(podStdOut).WithStderr(podStdErr).Run()
+
+	// Clean up and check the output
+	glooPodName := strings.Trim(podStdOut.String(), "'")
+	if glooPodName == "" {
+		fmt.Printf("Failed to get the name of the Gloo Gateway controller pod: %s\n", podStdErr.String())
+		return
+	}
+
+	// Get the metrics from the Gloo Gateway controller pod and write them to a file
+	metricsFilePath := filepath.Join(failureDir, "metrics.log")
+	metricsFile, err := os.OpenFile(metricsFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, os.ModePerm)
+	i.Assertions.Require.NoError(err)
+
+	// Using an ephemeral debug pod fetch the metrics from the Gloo Gateway controller
+	metricsCmd := i.Actions.Kubectl().Command(ctx, "debug", "-n", i.Metadata.InstallNamespace,
+		"-it", "--image=curlimages/curl:7.83.1", glooPodName, "--",
+		"curl", "http://localhost:9091/metrics")
+	_ = metricsCmd.WithStdout(metricsFile).WithStderr(metricsFile).Run()
+	metricsFile.Close()
+
+	// Get krt snapshot from the Gloo Gateway controller pod and write it to a file
+	krtSnapshotFilePath := filepath.Join(failureDir, "krt_snapshot.log")
+	krtSnapshotFile, err := os.OpenFile(krtSnapshotFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, os.ModePerm)
+	i.Assertions.Require.NoError(err)
+
+	// Using an ephemeral debug pod fetch the krt snapshot from the Gloo Gateway controller
+	krtSnapshotCmd := i.Actions.Kubectl().Command(ctx, "debug", "-n", i.Metadata.InstallNamespace,
+		"-it", "--image=curlimages/curl:7.83.1", glooPodName, "--",
+		"curl", "http://localhost:9095/snapshots/krt")
+	_ = krtSnapshotCmd.WithStdout(krtSnapshotFile).WithStderr(krtSnapshotFile).Run()
+	krtSnapshotFile.Close()
```

Comment:
while these make sense it would be ideal if we could add them to the main admin gloo endpoint tooling as its a pretty general case. I know its annoying to do it on a first pass (ie one case) but Im pretty sure I would love to have it on all cases

https://github.com/solo-io/gloo/blob/main/test/helpers/kube_dump.go#L46 is an example of our end dump and it includes hitting the stats endpoint like https://github.com/solo-io/gloo/blob/main/test/helpers/kube_dump.go#L311 which would be a nice setup that we can add to these types of tests